app/src/main/java/tech/ula/utils/FilesystemUtility.kt

```diff
@@ -81,19 +84,12 @@ class FilesystemUtility(
         }
     }
 
-    fun deleteFilesystem(filesystemId: Long) {
+    suspend fun deleteFilesystem(filesystemId: Long) {
         val directory = File("$applicationFilesDirPath/$filesystemId")
-        GlobalScope.launch {
-            if (directory.exists() && directory.isDirectory)
-                directory.deleteRecursively()
-            val isDirectoryDeleted = directory.deleteRecursively()
-            if (isDirectoryDeleted) {
-                val successMessage = "Successfully deleted filesystem located at: $directory"
-                logger.v("Filesystem Delete", successMessage)
-            } else {
-                val errorMessage = "Error in attempting to delete filesystem located at: $directory"
-                logger.e("Filesystem Delete", errorMessage)
-            }
+        if (!directory.exists() || !directory.isDirectory) return
```

Comment:
Can we change the `directory` variable to be called `filesystem` instead, it would make it easier to read.
For example: If the filesystem doesn't exist or the filesystem isn't a directory, then return.